commands/GitHub_Pattern_To_Repo_Export.php

```diff
@@ -0,0 +1,226 @@
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function initialize( InputInterface $input, OutputInterface $output ): void {
+
+		// Retrieve the given site.
+		$this->pressable_site = get_pressable_site_input( $input, $output, fn() => $this->prompt_site_input( $input, $output ) );
+		$output->writeln( "<info>Site {$this->pressable_site->id}: {$this->pressable_site->url}</info>", Output::VERBOSITY_VERBOSE );
+
+		// Store the ID of the site in the argument field.
+		$input->setArgument( 'site', $this->pressable_site->id );
+
+		// Check if the pattern name was already provided as an argument. If not, prompt the user for it.
+		$this->pattern_name = get_string_input( $input, $output, 'pattern-name', fn() => $this->prompt_pattern_name_input( $input, $output ));
+		$output->writeln( "<info>Pattern name: {$this->pattern_name}</info>", Output::VERBOSITY_VERBOSE );
+
+		// Check if the category slug was already provided as an argument. If not, prompt the user for it.
+		$this->category_slug = slugify( get_string_input( $input, $output, 'category-slug', fn() => $this->prompt_category_slug_input( $input, $output ) ) );
+		$output->writeln( "<info>Category slug: {$this->category_slug}</info>", Output::VERBOSITY_VERBOSE );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function execute( InputInterface $input, OutputInterface $output ): int {
+
+		$output->writeln( "<fg=magenta;options=bold>Exporting {$this->pattern_name} (Category: {$this->category_slug}) from {$this->pressable_site->displayName} (ID {$this->pressable_site->id}, URL {$this->pressable_site->url})</>" );
+
+		// Upload script.
+		$sftp_connection   = \Pressable_Connection_Helper::get_sftp_connection( $this->pressable_site->id );
+		if ( \is_null( $sftp_connection ) ) {
+			$output->writeln( '<error>Could not open SFTP connection.</error>' );
+			return Command::FAILURE;
+		}
+		$result = $sftp_connection->put( '/htdocs/pattern-extract.php', file_get_contents( __DIR__ . '/../scaffold/pattern-extract.php' ) );
+		if ( ! $result ) {
+			$output->writeln( "<error>Failed to copy pattern-extract.php to {$this->pressable_site->id}.</error>" );
+		}
+
+		$ssh_connection = \Pressable_Connection_Helper::get_ssh_connection( $this->pressable_site->id );
+		if ( \is_null( $ssh_connection ) ) {
+			$output->writeln( "<error>Failed to connect via SSH for {$this->pressable_site->url}. Aborting!</error>" );
+			return Command::FAILURE;
+		}
+
+		// Run script.
+		$result = $ssh_connection->exec( sprintf( "wp eval-file /htdocs/pattern-extract.php %s", escapeshellarg( $this->pattern_name ) ) );
+		$output->writeln( "<info>Pattern extraction result: {$result}</info>", Output::VERBOSITY_DEBUG );
+
+		// Delete script.
+		// $ssh_connection->exec( 'rm /htdocs/pattern-extract.php' );
+
+		if ( ! empty( $result ) ) {
+
+			// Temporary directory to clone the repository
+			$temp_dir = sys_get_temp_dir() . '/team51-patterns';
```

Comment:
Do you think we should use a random folder name? If something goes wrong here, the `team51-patterns` directory won't be deleted. And then the command will fail in the future because the folder already exists. Maybe we just append a randomly generated string at the end? 

Like:

```
$temp_dir = sys_get_temp_dir() . DIRECTORY_SEPARATOR . uniqid( 'team51-patterns_', true );
```

Reply:
That's a good idea. Added in a81da8f